### DISCUSSION TITLE
DocBlock in Blade to link to the used method

### DISCUSSION THREAD

**Discussion by ivandokov** [spatie/laravel-view-models#40] [Q&A] [answered]
I am wondering if there is any way to let the IDE know that a variable in blade (e.g. `$formatPrice(...)`) is actually using the ViewModel method `public function formatPrice(...)`. 
It will also help with refactoring the method I guess.

I've tried using `@uses` but its not really doing what I want.
Any ideas?

**Comment by brendt** [comment, marked as answer]
Referencing the function itself with its signature won't be possible. You could type hint the view model itself, but you'd need to call the function on it if you want proper static analysis:

```blade
@php
/** @var \App\…\PostViewModel $viewModel */
@endphp

{{ $viewModel->formatPrice($post) }}
```

**Reply by ivandokov** [reply]
That totally works. I didn't know we have access to the `$viewModel` instance. Thank you @brendt!
